torchmetrics/timeseries.py

```diff
@@ -0,0 +1,48 @@
+from typing import List, Tuple, Union
+
+import torch
+from torch import nn
+
+from torchmetrics.metric import Metric
+
+
+class TimeSeriesMetric(nn.ModuleList):
+    """
+    A class that keeps track of metrics as a timeseries, and implements useful methods.
+
+    Args:
+        metric: the torchmetric to keep track of at each timestep.
+    """
+
+    def __init__(self, metric: Metric):
+        super().__init__()
+        self.metric = metric
+
+    @property
+    def n_timesteps(self) -> int:
+        return len(self)
+
+    def add_timestep(self, preds: torch.Tensor, targets: torch.Tensor) -> None:
+        """Adds a metric module to the module list"""
+        module = self.metric()
```

Comment:
assuming that users have passed in a instantiated metric, we probably need to do a deep copy here
```suggestion
        module = deepcopy(self.metric)
```

Reply:
I think that here the `TimeSeriesMetric.metric` shall me metric type not instance

Reply:
but then it won't have the past measures so it would be waiting of compute...